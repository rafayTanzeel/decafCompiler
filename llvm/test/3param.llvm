; ModuleID = 'C'

declare void @print_int(i32)

define i32 @main() {
entry:
  %calltmp = call i32 @foo(i32 1, i32 2, i1 true)
  ret i32 0
}

define i32 @foo(i32 %x, i32 %y, i1 %z) {
entry:
  %x1 = alloca i32
  store i32 %x, i32* %x1
  %y2 = alloca i32
  store i32 %y, i32* %y2
  %z3 = alloca i1
  store i1 %z, i1* %z3
  store i32 1, i32* %x1
  store i32 1, i32* %y2
  store i1 false, i1* %z3
  %x4 = load i32, i32* %x1
  call void @print_int(i32 %x4)
  %y5 = load i32, i32* %y2
  call void @print_int(i32 %y5)
  %z6 = load i1, i1* %z3
  %zexttmp = zext i1 %z6 to i32
  call void @print_int(i32 %zexttmp)
  ret i32 0
}
