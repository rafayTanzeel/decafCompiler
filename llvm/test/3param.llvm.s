	.text
	.file	"llvm/test/3param.llvm.bc"
	.globl	main
	.align	16, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# BB#0:                                 # %entry
	pushq	%rax
.Ltmp0:
	.cfi_def_cfa_offset 16
	movl	$1, %edi
	movl	$2, %esi
	movl	$1, %edx
	callq	foo
	xorl	%eax, %eax
	popq	%rcx
	retq
.Lfunc_end0:
	.size	main, .Lfunc_end0-main
	.cfi_endproc

	.globl	foo
	.align	16, 0x90
	.type	foo,@function
foo:                                    # @foo
	.cfi_startproc
# BB#0:                                 # %entry
	subq	$24, %rsp
.Ltmp1:
	.cfi_def_cfa_offset 32
	movl	%edi, 20(%rsp)
	movl	%esi, 16(%rsp)
	andl	$1, %edx
	movb	%dl, 15(%rsp)
	movl	$1, 20(%rsp)
	movl	$1, 16(%rsp)
	movb	$0, 15(%rsp)
	movl	20(%rsp), %edi
	callq	print_int
	movl	16(%rsp), %edi
	callq	print_int
	movzbl	15(%rsp), %edi
	andl	$1, %edi
	callq	print_int
	xorl	%eax, %eax
	addq	$24, %rsp
	retq
.Lfunc_end1:
	.size	foo, .Lfunc_end1-foo
	.cfi_endproc


	.section	".note.GNU-stack","",@progbits
