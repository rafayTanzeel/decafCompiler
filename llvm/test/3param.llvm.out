semantic error: assignment of different types
